modules/reactor/src/meshgenerators/CoreMeshGenerator.C

```diff
@@ -69,6 +71,9 @@ CoreMeshGenerator::CoreMeshGenerator(const InputParameters & parameters)
     if (getMeshProperty<std::string>("reactor_params_name", _inputs[i]) != reactor_params)
       mooseError("The name of all reactor_params objects should be identical across all pins in "
                  "the input assemblies.\n");
+    if (getMeshProperty<bool>("homogenized_assembly", _inputs[i]) != assembly_homogenization)
+      mooseError(
+          "All assemblies in the core must be homogenized if assembly homogenization is used\n");
   }
```

Comment:
I've heard a few times people wanting to do mixed fidelity, this could be limiting. how hard would it be to relax this?

Reply:
Thanks for idea and comment. I brought this back to the team for consideration. We decided that it would be best for now to constrain the available cases in RGMB to either fully homogenized or fully heterogeneous, due to the current complexity in stitching together meshes whose number of nodes does not match. We would need to add a new mesh generator to create homogenized assemblies with nodes matching another mesh in order for this to be feasible. We would like to to be able to facilitate stitching of complicated non matching nodes for the user but for now this isn't practical yet with the current toolset. Therefore, we'd like to postpone allowing the mixing and matching of hom/het assemblies til a later date.

Reply:
sounds good